services/user.js

```diff
@@ -377,14 +381,45 @@ class User {
         }
 
         updatedUser.dataCommons = DataCommon.get(user[0]?.role, user[0]?.dataCommons, params?.role, params?.dataCommons);
-        // Check if an organization is required and missing for the user's role
-        const userHasOrg = Boolean(user[0]?.organization?.orgID);
-        if (!userHasOrg && [USER.ROLES.DC_POC, USER.ROLES.ORG_OWNER, USER.ROLES.SUBMITTER, USER.ROLES.FEDERAL_MONITOR].includes(updatedUser.role || user[0]?.role)) {
-            throw new Error(ERROR.USER_ORG_REQUIRED);
+        // add studies to user.
+        const validStudies = await this.#findApprovedStudies(params?.studies);
+        if (params?.studies && params.studies.length > 0) {
+            if(validStudies.length !== params.studies.length) {
+                throw new Error(ERROR.INVALID_NOT_APPROVED_STUDIES);
+            }
+            else {
+                updatedUser.studies = (params.studies[0] instanceof Object)?params.studies:params.studies.map(str => ({ _id: str }));
+            }
+        }
+        else
+            updatedUser.studies = [];
+
+        if (params?.status){
+            if (! [USER.STATUSES.ACTIVE, USER.STATUSES.INACTIVE].includes(params.status))
+                throw new Error(ERROR.INVALID_USER_STATUS);
+
+            updatedUser.status = params.status
+        }
+
+        const res = await this.userCollection.findOneAndUpdate({ _id: params.userID }, {...updatedUser, updateAt: getCurrentTime()}, {returnDocument: 'after'});
+        const userAfterUpdate = res.value;
+        const prevUser = user[0];
+        if (userAfterUpdate) {
+            const promiseArray = [
+                await this.#notifyDeactivatedUser(prevUser, userAfterUpdate.status),
+                await this.#notifyUpdatedUser(prevUser, userAfterUpdate, userAfterUpdate.role),
+                await this.#logAfterUserEdit(prevUser, userAfterUpdate)
+            ];
+            await Promise.all(promiseArray);
+        } else {
+            throw new Error(ERROR.UPDATE_FAILED);
         }
-        return await this.updateUserInfo(user[0], updatedUser, params.userID, params.status, params.role, params?.studies);
```

Comment:
The codes are duplicates. Could you update the updateUserInfo instead of adding new? I will update changes later after PR merge.

Reply:
updated.